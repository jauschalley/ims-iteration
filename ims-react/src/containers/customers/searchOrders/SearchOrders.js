import React from 'react';
import {observer} from "mobx-react";
import {Formik, Field} from 'formik';
import {Row, Container, Col, Form, FormLabel, Button} from 'react-bootstrap';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import CwigCard from '../../../components/CwigCard';

const SearchOrders = observer(({orderSearchStore, referenceDataStore}) => {
  let refData = referenceDataStore.referenceData;

  const mapSelectOptions = (options, labelKey, idKey) => (
    options.map((option) => (
      <option value={option[idKey]}>
        {option[labelKey]}
      </option>
    ))
  );

  return(
    <Container>
      <Row><h1 class="margin-30">Search Orders</h1></Row>
      <Formik initialValues={{
        orderStatus:'',
        orderType: '',
        orderId: '',
        customerFirstName: '',
        customerLastName: '',
        zipCode: '',
        dateFrom: '',
        dateTo: ''
      }}
      onSubmit={(values) => orderSearchStore.searchOrders(values)}
      >
        {formikProps =>
          <Form>
            <CwigCard>
              <Row>
                <Col md={4}>
                  <FormLabel>Order Status</FormLabel>
                  <Field as="select" name="orderStatus">
                    {mapSelectOptions(refData.orderStatuses, 'status', 'id')}
                  </Field>
                </Col>
                <Col md={4}>
                  <FormLabel>Order Type</FormLabel>
                  <Field as="select" name="orderStatus">
                    {mapSelectOptions(refData.orderTypes, 'orderType', 'typeID')}
                  </Field>
                </Col>
                <Col md={4}>
                  <FormLabel>Order ID</FormLabel>
                  <Field placeholder="Order ID" name="orderId"></Field>
                </Col>
              </Row>
              <Row>
                <Col md={4}>
                  <FormLabel>Customer First Name</FormLabel>
                  <Field placeholder="First Name" name="customerFirstName"></Field>
                </Col>
                <Col md={4}>
                  <FormLabel>Customer Last Name</FormLabel>
                  <Field placeholder="Last Name" name="customerLastName"></Field>
                </Col>
                <Col md={4}>
                  <FormLabel>Customer Zip Code</FormLabel>
                  <Field placeholder="Zip Code" name="zipCode"></Field>
                </Col>
              </Row>
              <Row>
                <Col md={6}>
                  <FormLabel>Date Range</FormLabel>
                  <Field placeholder="Date From" name="dateFrom"></Field>
                  <Field placeholder="Date To" name="dateTo"></Field>
                </Col>
              </Row>
              <Row>
                <Button className="search" onClick={formikProps.handleSubmit}>Search</Button>
              </Row>
            </CwigCard>
          </Form>
        }
      </Formik>
      {orderSearchStore.searchResults && <TableContainer component={Paper}>
        <Table aria-label="search customers table">
          <TableHead>
            <TableRow>
              <TableCell>Order ID</TableCell>
              <TableCell>Order Date</TableCell>
              <TableCell>Customer</TableCell>
              <TableCell>Status</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {orderSearchStore.searchResults.map(row => (
              <TableRow key={row.name}>
                <TableCell>{row.orderID}</TableCell>
                <TableCell>{row.orderDate}</TableCell>
                <TableCell>{row.firstName} {row.lastNameName}</TableCell>
                <TableCell>{row.status}</TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </TableContainer>}
    </Container>
  );
});

export default SearchOrders;