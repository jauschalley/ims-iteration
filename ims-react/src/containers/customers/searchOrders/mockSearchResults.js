const searchResults = {"testmode":"0","orders":[{"processedDate":"","orderType":"Customer","takenByImsUserId":50,"firstName":"","items":[],"orderID":488773,"statusID":1,"status":"Pending","customerID":101722,"shippingLastName":"","shippingZip":"","orderDate":"February, 26 2020 00:00:00","lastName":"","typeID":1,"shippingFirstName":""},{"processedDate":"","orderType":"Customer","takenByImsUserId":50,"firstName":"","items":[],"orderID":488772,"statusID":1,"status":"Pending","customerID":101721,"shippingLastName":"","shippingZip":"","orderDate":"February, 26 2020 00:00:00","lastName":"","typeID":1,"shippingFirstName":""},{"processedDate":"","orderType":"Customer","takenByImsUserId":50,"firstName":"Dr. Edwina","items":[],"orderID":488771,"statusID":1,"status":"Pending","customerID":101719,"shippingLastName":"Eden","shippingZip":"58921","orderDate":"February, 26 2020 00:00:00","lastName":"Eden","typeID":1,"shippingFirstName":"Dr. Edwina"},{"processedDate":"","orderType":"Customer","takenByImsUserId":50,"firstName":"Geraldine","items":[],"orderID":488770,"statusID":1,"status":"Pending","customerID":101716,"shippingLastName":"George","shippingZip":"38764","orderDate":"February, 26 2020 00:00:00","lastName":"George","typeID":1,"shippingFirstName":"Geraldine"},{"processedDate":"","orderType":"Customer","takenByImsUserId":50,"firstName":"","items":[],"orderID":488769,"statusID":1,"status":"Pending","customerID":101710,"shippingLastName":"","shippingZip":"","orderDate":"February, 25 2020 00:00:00","lastName":"","typeID":1,"shippingFirstName":""},{"processedDate":"","orderType":"Customer","takenByImsUserId":100,"firstName":"","items":[],"orderID":488768,"statusID":1,"status":"Pending","customerID":101696,"shippingLastName":"","shippingZip":"","orderDate":"February, 20 2020 00:00:00","lastName":"","typeID":1,"shippingFirstName":""},{"processedDate":"","orderType":"Customer","takenByImsUserId":50,"firstName":"Charles","items":[],"orderID":488767,"statusID":1,"status":"Pending","customerID":101707,"shippingLastName":"Crowford","shippingZip":"28945","orderDate":"February, 17 2020 00:00:00","lastName":"Crowford","typeID":1,"shippingFirstName":"Charles"},{"processedDate":"","orderType":"Customer","takenByImsUserId":50,"firstName":"Grace","items":[],"orderID":488766,"statusID":1,"status":"Pending","customerID":101703,"shippingLastName":"Chan","shippingZip":"38912","orderDate":"February, 17 2020 00:00:00","lastName":"Chan","typeID":1,"shippingFirstName":"Grace"},{"processedDate":"","orderType":"Customer","takenByImsUserId":50,"firstName":"Charlotte","items":[],"orderID":488765,"statusID":1,"status":"Pending","customerID":101698,"shippingLastName":"Grayson","shippingZip":"49812","orderDate":"February, 17 2020 00:00:00","lastName":"Grayson","typeID":1,"shippingFirstName":"Charlotte"},{"processedDate":"","orderType":"Customer","takenByImsUserId":50,"firstName":"","items":[],"orderID":488764,"statusID":1,"status":"Pending","customerID":101680,"shippingLastName":"","shippingZip":"","orderDate":"February, 17 2020 00:00:00","lastName":"","typeID":1,"shippingFirstName":""},{"processedDate":"","orderType":"Customer","takenByImsUserId":50,"firstName":"Jocelyn","items":[],"orderID":488763,"statusID":1,"status":"Pending","customerID":101693,"shippingLastName":"Brown","shippingZip":"62629","orderDate":"February, 14 2020 00:00:00","lastName":"Brown","typeID":1,"shippingFirstName":"Jocelyn"},{"processedDate":"","orderType":"Customer","takenByImsUserId":50,"firstName":"Mickey","items":[],"orderID":488762,"statusID":1,"status":"Pending","customerID":101690,"shippingLastName":"Mouse","shippingZip":"27653","orderDate":"February, 14 2020 00:00:00","lastName":"Mouse","typeID":1,"shippingFirstName":"Mickey"},{"processedDate":"","orderType":"Customer","takenByImsUserId":100,"firstName":"","items":[],"orderID":488761,"statusID":1,"status":"Pending","customerID":101685,"shippingLastName":"","shippingZip":"","orderDate":"February, 13 2020 00:00:00","lastName":"","typeID":1,"shippingFirstName":""},{"processedDate":"","orderType":"Customer","takenByImsUserId":50,"firstName":"Marilena","items":[],"orderID":488760,"statusID":1,"status":"Pending","customerID":101684,"shippingLastName":"Queen","shippingZip":"25786","orderDate":"February, 13 2020 00:00:00","lastName":"Queen","typeID":1,"shippingFirstName":"Marilena"},{"processedDate":"","orderType":"Customer","takenByImsUserId":50,"firstName":"Alice","items":[],"orderID":488759,"statusID":1,"status":"Pending","customerID":101679,"shippingLastName":"Smith","shippingZip":"13324","orderDate":"February, 13 2020 00:00:00","lastName":"Smith","typeID":1,"shippingFirstName":"Alice"},{"processedDate":"","orderType":"Customer","takenByImsUserId":50,"firstName":"","items":[],"orderID":488758,"statusID":1,"status":"Pending","customerID":66868,"shippingLastName":"Cooper-Smith","shippingZip":"","orderDate":"February, 12 2020 00:00:00","lastName":"Cooper-Smith","typeID":1,"shippingFirstName":""},{"processedDate":"","orderType":"Customer","takenByImsUserId":50,"firstName":"Geraldine","items":[],"orderID":488757,"statusID":1,"status":"Pending","customerID":101670,"shippingLastName":"Lee","shippingZip":"58912","orderDate":"February, 10 2020 00:00:00","lastName":"Lee","typeID":1,"shippingFirstName":"Geraldine"},{"processedDate":"","orderType":"Customer","takenByImsUserId":50,"firstName":"Belle","items":[],"orderID":488756,"statusID":1,"status":"Pending","customerID":101666,"shippingLastName":"Aire ","shippingZip":"25890","orderDate":"February, 10 2020 00:00:00","lastName":"Aire ","typeID":1,"shippingFirstName":"Belle"},{"processedDate":"","orderType":"Customer","takenByImsUserId":50,"firstName":"Crystal","items":[],"orderID":488755,"statusID":1,"status":"Pending","customerID":101664,"shippingLastName":"Clear","shippingZip":"23010","orderDate":"February, 10 2020 00:00:00","lastName":"Clear","typeID":1,"shippingFirstName":"Crystal"},{"processedDate":"","orderType":"Customer","takenByImsUserId":50,"firstName":"","items":[],"orderID":488754,"statusID":1,"status":"Pending","customerID":101660,"shippingLastName":"","shippingZip":"","orderDate":"February, 10 2020 00:00:00","lastName":"","typeID":1,"shippingFirstName":""},{"processedDate":"","orderType":"Customer","takenByImsUserId":50,"firstName":"Bob","items":[],"orderID":488753,"statusID":1,"status":"Pending","customerID":101656,"shippingLastName":"Lee","shippingZip":"22031","orderDate":"February, 06 2020 00:00:00","lastName":"Lee","typeID":1,"shippingFirstName":"Bob"},{"processedDate":"","orderType":"Online","takenByImsUserId":36,"firstName":"","items":[{"orderableProductID":3863,"calculatedSKU":"XPCN-20-10280-BD","quantity":1,"title":"Supervising Child Protective Services Caseworkers"}],"orderID":488752,"statusID":1,"status":"Pending","customerID":"","shippingLastName":"","shippingZip":"","orderDate":"February, 06 2020 11:59:33","lastName":"","typeID":3,"shippingFirstName":""},{"processedDate":"","orderType":"Online","takenByImsUserId":36,"firstName":"","items":[{"orderableProductID":5644,"calculatedSKU":"XPCN-11-11290-UB","quantity":1,"title":"Preventing Child Abuse and Neglect"}],"orderID":488751,"statusID":1,"status":"Pending","customerID":"","shippingLastName":"","shippingZip":"","orderDate":"February, 06 2020 09:30:14","lastName":"","typeID":3,"shippingFirstName":""},{"processedDate":"","orderType":"Online","takenByImsUserId":36,"firstName":"","items":[{"orderableProductID":5644,"calculatedSKU":"XPCN-11-11290-UB","quantity":1,"title":"Preventing Child Abuse and Neglect"}],"orderID":488750,"statusID":1,"status":"Pending","customerID":"","shippingLastName":"","shippingZip":"","orderDate":"February, 06 2020 09:30:12","lastName":"","typeID":3,"shippingFirstName":""},{"processedDate":"","orderType":"Online","takenByImsUserId":36,"firstName":"","items":[{"orderableProductID":5503,"calculatedSKU":"XPCN-AB-0003A-UB","quantity":1,"title":"The Adoption Home Study Process"}],"orderID":488749,"statusID":1,"status":"Pending","customerID":"","shippingLastName":"","shippingZip":"","orderDate":"February, 06 2020 08:53:39","lastName":"","typeID":3,"shippingFirstName":""},{"processedDate":"","orderType":"Online","takenByImsUserId":36,"firstName":"","items":[{"orderableProductID":5675,"calculatedSKU":"XPCN-AB-0047A-UB","quantity":1,"title":"Sibling Issues in Foster Care and Adoption"}],"orderID":488748,"statusID":1,"status":"Pending","customerID":"","shippingLastName":"","shippingZip":"","orderDate":"February, 06 2020 08:52:08","lastName":"","typeID":3,"shippingFirstName":""},{"processedDate":"","orderType":"Online","takenByImsUserId":36,"firstName":"","items":[{"orderableProductID":5643,"calculatedSKU":"XPCN-11-11291-UB","quantity":1,"title":"Leaving Your Child Home Alone"}],"orderID":488747,"statusID":1,"status":"Pending","customerID":"","shippingLastName":"","shippingZip":"","orderDate":"February, 06 2020 03:11:20","lastName":"","typeID":3,"shippingFirstName":""},{"processedDate":"","orderType":"Online","takenByImsUserId":36,"firstName":"","items":[{"orderableProductID":5643,"calculatedSKU":"XPCN-11-11291-UB","quantity":1,"title":"Leaving Your Child Home Alone"}],"orderID":488746,"statusID":1,"status":"Pending","customerID":"","shippingLastName":"","shippingZip":"","orderDate":"February, 06 2020 03:11:17","lastName":"","typeID":3,"shippingFirstName":""},{"processedDate":"","orderType":"Online","takenByImsUserId":36,"firstName":"","items":[{"orderableProductID":5668,"calculatedSKU":"XPCN-11-11294-UB","quantity":1,"title":"Los Hijos Solos en Casa (Leaving Your Child Home Alone)"}],"orderID":488745,"statusID":1,"status":"Pending","customerID":"","shippingLastName":"","shippingZip":"","orderDate":"February, 06 2020 02:47:40","lastName":"","typeID":3,"shippingFirstName":""},{"processedDate":"","orderType":"Online","takenByImsUserId":36,"firstName":"","items":[{"orderableProductID":5668,"calculatedSKU":"XPCN-11-11294-UB","quantity":1,"title":"Los Hijos Solos en Casa (Leaving Your Child Home Alone)"}],"orderID":488744,"statusID":1,"status":"Pending","customerID":"","shippingLastName":"","shippingZip":"","orderDate":"February, 06 2020 02:47:39","lastName":"","typeID":3,"shippingFirstName":""},{"processedDate":"","orderType":"Online","takenByImsUserId":36,"firstName":"","items":[{"orderableProductID":5644,"calculatedSKU":"XPCN-11-11290-UB","quantity":1,"title":"Preventing Child Abuse and Neglect"}],"orderID":488743,"statusID":1,"status":"Pending","customerID":"","shippingLastName":"","shippingZip":"","orderDate":"February, 06 2020 00:53:06","lastName":"","typeID":3,"shippingFirstName":""},{"processedDate":"","orderType":"Online","takenByImsUserId":36,"firstName":"","items":[],"orderID":488742,"statusID":1,"status":"Pending","customerID":"","shippingLastName":"","shippingZip":"","orderDate":"February, 06 2020 00:22:25","lastName":"","typeID":3,"shippingFirstName":""},{"processedDate":"","orderType":"Online","takenByImsUserId":36,"firstName":"","items":[{"orderableProductID":5679,"calculatedSKU":"XPCN-AB-0062A-UB","quantity":1,"title":"Adopting as a Single Parent"}],"orderID":488741,"statusID":1,"status":"Pending","customerID":"","shippingLastName":"","shippingZip":"","orderDate":"February, 05 2020 22:37:41","lastName":"","typeID":3,"shippingFirstName":""},{"processedDate":"","orderType":"Online","takenByImsUserId":36,"firstName":"","items":[{"orderableProductID":5525,"calculatedSKU":"XPCN-11-11380-UB","quantity":1,"title":"Reunification: Bringing Your Children Home From Foster Care"},{"orderableProductID":5552,"calculatedSKU":"XPCN-11-11392-UB","quantity":1,"title":"Reunificación: trayendo a sus hijos de regreso a casa después del cuidado de crianza (Reunification: Bringing Your Children Home From Foster Care)"},{"orderableProductID":4706,"calculatedSKU":"XPCN-11-11300-UB","quantity":1,"title":"How the Child Welfare System Works"},{"orderableProductID":4785,"calculatedSKU":"XPCN-11-11330-UB","quantity":1,"title":"Cómo Funciona el Sistema de Bienestar de Menores (How the Child Welfare System Works)"},{"orderableProductID":5524,"calculatedSKU":"XPCN-AB-0026A-UB","quantity":1,"title":"Kinship Caregivers and the Child Welfare System"},{"orderableProductID":5651,"calculatedSKU":"XPCN-AB-0026E-UB","quantity":1,"title":"Parientes como proveedores de cuidado y el sistema de bienestar de menores (Kinship Caregivers and the Child Welfare System)"},{"orderableProductID":5677,"calculatedSKU":"XPCN-AB-0077A-UB","quantity":1,"title":"Families Considering Foster Care and Adoption"},{"orderableProductID":5609,"calculatedSKU":"XPCN-AB-0072A-UB","quantity":1,"title":"Helping Your Child Transition from Foster Care to Adoption"},{"orderableProductID":5681,"calculatedSKU":"XPCN-AB-0079A-UB","quantity":1,"title":"Belonging Matters—Helping Youth Explore Permanency"},{"orderableProductID":5637,"calculatedSKU":"XPCN-AB-0073A-UB","quantity":1,"title":"Finding and Working with Adoption-Competent Therapists"},{"orderableProductID":5209,"calculatedSKU":"XPCN-11-11352-UB","quantity":1,"title":"Preparing Children and Youth for Adoption or Other Family Permanency"},{"orderableProductID":5654,"calculatedSKU":"XPCN-11-11412-UB","quantity":1,"title":"Promoting Permanency for Older Youth in Out-of-Home Care"},{"orderableProductID":5590,"calculatedSKU":"XPCN-AB-0070A-UB","quantity":1,"title":"Providing Adoption Support and Preservation Services"},{"orderableProductID":5675,"calculatedSKU":"XPCN-AB-0047A-UB","quantity":1,"title":"Sibling Issues in Foster Care and Adoption"},{"orderableProductID":5563,"calculatedSKU":"XPCN-11-11318-UB","quantity":1,"title":"Supporting Brain Development in Traumatized Children and Youth"},{"orderableProductID":5321,"calculatedSKU":"XPCN-AB-0034A-UB","quantity":1,"title":"Are You Pregnant and Thinking About Adoption?"},{"orderableProductID":5338,"calculatedSKU":"XPCN-AB-0034I-UB","quantity":1,"title":"¿Está Embarazada y Pensando en la Adopción? (Are You Pregnant and Thinking About Adoption?)"},{"orderableProductID":5099,"calculatedSKU":"XPCN-AB-0031A-UB","quantity":1,"title":"Impact of Adoption on Birth Parents"},{"orderableProductID":5220,"calculatedSKU":"XPCN-AB-0031I-UB","quantity":1,"title":"El Impacto de la Adopción en los Padres Biológicos (Impact of Adoption on Birth Parents)"}],"orderID":488740,"statusID":1,"status":"Pending","customerID":"","shippingLastName":"","shippingZip":"","orderDate":"February, 05 2020 23:10:53","lastName":"","typeID":3,"shippingFirstName":""},{"processedDate":"","orderType":"Online","takenByImsUserId":36,"firstName":"","items":[{"orderableProductID":5394,"calculatedSKU":"XPCN-AB-0063A-UB","quantity":1,"title":"Intercountry Adoption From Hague Convention and Non-Hague Convention Countries"},{"orderableProductID":5380,"calculatedSKU":"XPCN-AB-0009A-UB","quantity":1,"title":"Intercountry Adoption: Where Do I Start?"},{"orderableProductID":5493,"calculatedSKU":"XPCN-AB-0065A-UB","quantity":1,"title":"Adoption Options: Where Do I Start?"}],"orderID":488739,"statusID":1,"status":"Pending","customerID":"","shippingLastName":"","shippingZip":"","orderDate":"February, 05 2020 20:58:31","lastName":"","typeID":3,"shippingFirstName":""},{"processedDate":"","orderType":"Online","takenByImsUserId":36,"firstName":"","items":[],"orderID":488738,"statusID":1,"status":"Pending","customerID":"","shippingLastName":"","shippingZip":"","orderDate":"February, 05 2020 18:53:56","lastName":"","typeID":3,"shippingFirstName":""},{"processedDate":"","orderType":"Online","takenByImsUserId":36,"firstName":"","items":[],"orderID":488737,"statusID":1,"status":"Pending","customerID":"","shippingLastName":"","shippingZip":"","orderDate":"February, 05 2020 18:53:54","lastName":"","typeID":3,"shippingFirstName":""},{"processedDate":"","orderType":"Online","takenByImsUserId":36,"firstName":"","items":[],"orderID":488736,"statusID":1,"status":"Pending","customerID":"","shippingLastName":"","shippingZip":"","orderDate":"February, 05 2020 18:53:50","lastName":"","typeID":3,"shippingFirstName":""},{"processedDate":"","orderType":"Online","takenByImsUserId":36,"firstName":"","items":[],"orderID":488735,"statusID":1,"status":"Pending","customerID":"","shippingLastName":"","shippingZip":"","orderDate":"February, 05 2020 18:53:48","lastName":"","typeID":3,"shippingFirstName":""},{"processedDate":"","orderType":"Online","takenByImsUserId":36,"firstName":"","items":[{"orderableProductID":4377,"calculatedSKU":"XPCN-AJ-0008A-UB","quantity":1,"title":"Adoption Disruption and Dissolution"}],"orderID":488734,"statusID":1,"status":"Pending","customerID":"","shippingLastName":"","shippingZip":"","orderDate":"February, 05 2020 18:31:24","lastName":"","typeID":3,"shippingFirstName":""},{"processedDate":"","orderType":"Online","takenByImsUserId":36,"firstName":"","items":[],"orderID":488733,"statusID":1,"status":"Pending","customerID":"","shippingLastName":"","shippingZip":"","orderDate":"February, 05 2020 18:28:19","lastName":"","typeID":3,"shippingFirstName":""},{"processedDate":"","orderType":"Online","takenByImsUserId":36,"firstName":"","items":[],"orderID":488732,"statusID":1,"status":"Pending","customerID":"","shippingLastName":"","shippingZip":"","orderDate":"February, 05 2020 18:28:17","lastName":"","typeID":3,"shippingFirstName":""},{"processedDate":"February, 06 2020 00:00:00","orderType":"Online","takenByImsUserId":36,"firstName":"Slvdefhwr","items":[{"orderableProductID":5523,"calculatedSKU":"XPCN-11-11379-UB","quantity":1,"title":"From Complaint to Resolution: Understanding the Child Welfare Grievance Process"},{"orderableProductID":5098,"calculatedSKU":"XPCN-AB-0032A-UB","quantity":1,"title":"Impact of Adoption on Adopted Persons"},{"orderableProductID":5099,"calculatedSKU":"XPCN-AB-0031A-UB","quantity":1,"title":"Impact of Adoption on Birth Parents"},{"orderableProductID":5653,"calculatedSKU":"XPCN-11-11413-UB","quantity":1,"title":"Partnering With Birth Parents To Promote Reunification"},{"orderableProductID":5525,"calculatedSKU":"XPCN-11-11380-UB","quantity":1,"title":"Reunification: Bringing Your Children Home From Foster Care"},{"orderableProductID":5541,"calculatedSKU":"XPCN-11-11316-UB","quantity":1,"title":"Understanding Child Welfare and the Courts"},{"orderableProductID":5531,"calculatedSKU":"XPCN-11-11309-UB","quantity":1,"title":"Caseload and Workload Management"},{"orderableProductID":5537,"calculatedSKU":"XPCN-11-11385-UB","quantity":1,"title":"Family Engagement: Partnering With Families to Improve Child Welfare Outcomes"},{"orderableProductID":5649,"calculatedSKU":"XPCN-11-11411-UB","quantity":1,"title":"Parent Education to Strengthen Families and Prevent Child Maltreatment"},{"orderableProductID":5646,"calculatedSKU":"XPCN-11-11409-UB","quantity":1,"title":"Parent-Child Interaction Therapy: A Primer for Child Welfare Professionals"},{"orderableProductID":5589,"calculatedSKU":"XPCN-11-11332-UB","quantity":1,"title":"Rural Child Welfare Practice"},{"orderableProductID":5572,"calculatedSKU":"XPCN-11-11398-UB","quantity":1,"title":"Supporting Successful Reunifications"},{"orderableProductID":5630,"calculatedSKU":"XPCN-11-11406-UB","quantity":1,"title":"Building Partnerships to Support Stable Housing for Child Welfare-Involved Families and Youth"},{"orderableProductID":5615,"calculatedSKU":"XPCN-11-11404-UB","quantity":1,"title":"Concurrent Planning for Timely Permanence"},{"orderableProductID":3182,"calculatedSKU":"XPCN-11-11251-BD","quantity":1,"title":"Grants.gov : one site, every grant opportunity / Department of Health and Human Services"},{"orderableProductID":5542,"calculatedSKU":"XPCN-39-40018-BD","quantity":1,"title":"Building Community Building Hope: 2016/2017 Prevention Resource Guide"},{"orderableProductID":4129,"calculatedSKU":"XPCN-20-10334-BD","quantity":1,"title":"Drug Testing in Child Welfare: Practice and Policy Considerations."},{"orderableProductID":4706,"calculatedSKU":"XPCN-11-11300-UB","quantity":1,"title":"How the Child Welfare System Works"},{"orderableProductID":5639,"calculatedSKU":"XPCN-11-11407-UB","quantity":1,"title":"Parenting Children and Youth Who Have Experienced Abuse or Neglect"},{"orderableProductID":3890,"calculatedSKU":"XPCN-20-10310-BD","quantity":1,"title":"Screening and Assessment for Family Engagement, Retention and Recovery (SAFERR)"},{"orderableProductID":3863,"calculatedSKU":"XPCN-20-10280-BD","quantity":1,"title":"Supervising Child Protective Services Caseworkers"},{"orderableProductID":3586,"calculatedSKU":"XPCN-20-10300-BD","quantity":1,"title":"Synthesis of Findings : Substance Abuse Child Welfare Waiver Demonstrations."},{"orderableProductID":5517,"calculatedSKU":"XPCN-20-10352-BD","quantity":1,"title":"Within Our Reach: A National Strategy to Eliminate Child Abuse and Neglect Fatalities"}],"orderID":488731,"statusID":4,"status":"Processed","customerID":101653,"shippingLastName":"Henry","shippingZip":"72616","orderDate":"February, 05 2020 18:14:23","lastName":"Henry","typeID":3,"shippingFirstName":"Npfstcrmd"},{"processedDate":"","orderType":"Online","takenByImsUserId":36,"firstName":"","items":[],"orderID":488730,"statusID":1,"status":"Pending","customerID":"","shippingLastName":"","shippingZip":"","orderDate":"February, 05 2020 17:38:57","lastName":"","typeID":3,"shippingFirstName":""},{"processedDate":"","orderType":"Online","takenByImsUserId":36,"firstName":"","items":[],"orderID":488729,"statusID":1,"status":"Pending","customerID":"","shippingLastName":"","shippingZip":"","orderDate":"February, 05 2020 17:38:55","lastName":"","typeID":3,"shippingFirstName":""},{"processedDate":"","orderType":"Online","takenByImsUserId":36,"firstName":"","items":[],"orderID":488728,"statusID":1,"status":"Pending","customerID":"","shippingLastName":"","shippingZip":"","orderDate":"February, 05 2020 17:33:39","lastName":"","typeID":3,"shippingFirstName":""},{"processedDate":"","orderType":"Online","takenByImsUserId":36,"firstName":"","items":[],"orderID":488727,"statusID":1,"status":"Pending","customerID":"","shippingLastName":"","shippingZip":"","orderDate":"February, 05 2020 17:33:37","lastName":"","typeID":3,"shippingFirstName":""},{"processedDate":"","orderType":"Online","takenByImsUserId":36,"firstName":"","items":[],"orderID":488726,"statusID":1,"status":"Pending","customerID":"","shippingLastName":"","shippingZip":"","orderDate":"February, 05 2020 16:44:27","lastName":"","typeID":3,"shippingFirstName":""},{"processedDate":"","orderType":"Online","takenByImsUserId":36,"firstName":"","items":[],"orderID":488725,"statusID":1,"status":"Pending","customerID":"","shippingLastName":"","shippingZip":"","orderDate":"February, 05 2020 16:44:22","lastName":"","typeID":3,"shippingFirstName":""},{"processedDate":"","orderType":"Online","takenByImsUserId":36,"firstName":"","items":[],"orderID":488724,"statusID":1,"status":"Pending","customerID":"","shippingLastName":"","shippingZip":"","orderDate":"February, 05 2020 16:40:18","lastName":"","typeID":3,"shippingFirstName":""}],"status_code":"200","status_text":""}

export default searchResults;