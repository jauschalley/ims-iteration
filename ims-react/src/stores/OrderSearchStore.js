import {observable, action, decorate} from "mobx";
import RequestService from '../services/RequestService';
import config from '../config';

class OrderSearchStore {
  searchResults = null;

  searchOrders = (formData) => {
    RequestService.post(`${config.backendUrl}/orders`, formData, (response) => {
      this.searchResults = response.data.orders;
    });
  }
}

decorate(OrderSearchStore, {
  searchResults: observable,
  searchOrders: action
});

export default OrderSearchStore;