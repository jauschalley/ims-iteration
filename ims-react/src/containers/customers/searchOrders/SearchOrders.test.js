import React from 'react';
import renderer from 'react-test-renderer';
import SearchOrders from './SearchOrders';
import referenceData from '../newInquiry/mockReferenceData';
import mockSearchResults from './mockSearchResults';
import ReferenceDataStore from '../../../stores/ReferenceDataStore';
import OrderSearchStore from '../../../stores/OrderSearchStore';

describe('<SearchOrders />', () => {
  it('renders correctly', () => {
    const referenceDataStore = new ReferenceDataStore(referenceData);
    const orderSearchStore = new OrderSearchStore(mockSearchResults);
    const tree = renderer
      .create(
        <SearchOrders 
          orderSearchStore={orderSearchStore}
          referenceDataStore={referenceDataStore} 
        />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});