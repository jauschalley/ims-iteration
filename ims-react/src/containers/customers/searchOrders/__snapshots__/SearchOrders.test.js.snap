// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchOrders /> renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <h1
      class="margin-30"
    >
      Search Orders
    </h1>
  </div>
  <form
    className=""
  >
    <div
      className="cwig-card container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-4"
        >
          <label
            className="form-label"
          >
            Order Status
          </label>
          <select
            name="orderStatus"
            onBlur={[Function]}
            onChange={[Function]}
            value=""
          >
            <option
              value={1}
            >
              Pending
            </option>
            <option
              value={2}
            >
              Archived
            </option>
            <option
              value={3}
            >
              Confirmed
            </option>
            <option
              value={4}
            >
              Processed
            </option>
            <option
              value={5}
            >
              Held
            </option>
          </select>
        </div>
        <div
          className="col-md-4"
        >
          <label
            className="form-label"
          >
            Order Type
          </label>
          <select
            name="orderStatus"
            onBlur={[Function]}
            onChange={[Function]}
            value=""
          >
            <option
              value={2}
            >
              Conference
            </option>
            <option
              value={1}
            >
              Customer
            </option>
            <option
              value={3}
            >
              Online
            </option>
            <option
              value={4}
            >
              Pre-Order
            </option>
            <option
              value={5}
            >
              Targeted Mailing
            </option>
          </select>
        </div>
        <div
          className="col-md-4"
        >
          <label
            className="form-label"
          >
            Order ID
          </label>
          <input
            name="orderId"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Order ID"
            value=""
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-4"
        >
          <label
            className="form-label"
          >
            Customer First Name
          </label>
          <input
            name="customerFirstName"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="First Name"
            value=""
          />
        </div>
        <div
          className="col-md-4"
        >
          <label
            className="form-label"
          >
            Customer Last Name
          </label>
          <input
            name="customerLastName"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Last Name"
            value=""
          />
        </div>
        <div
          className="col-md-4"
        >
          <label
            className="form-label"
          >
            Customer Zip Code
          </label>
          <input
            name="zipCode"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Zip Code"
            value=""
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-6"
        >
          <label
            className="form-label"
          >
            Date Range
          </label>
          <input
            name="dateFrom"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Date From"
            value=""
          />
          <input
            name="dateTo"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Date To"
            value=""
          />
        </div>
      </div>
      <div
        className="row"
      >
        <button
          className="search btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Search
        </button>
      </div>
    </div>
  </form>
</div>
`;
